From 0c3398b1acf2a2cabd9cee91dc3fe3d35805fa8b Mon Sep 17 00:00:00 2001
From: Evgeni Golov <evgeni@golov.de>
Date: Fri, 16 Jun 2023 18:13:54 +0200
Subject: [PATCH] update DEFINE_SEMAPHORE usage for linux 6.4+

Linux changed the DEFINE_SEMAPHORE macro in
48380368dec14859723b9e3fbd43e042638d9a76, making it take two parameters
instead of one. Pass an explicit 1 as on 6.4+.
---
 thinkpad_ec.c | 4 +++-
 tp_smapi.c    | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/thinkpad_ec.c b/thinkpad_ec.c
index a8e812f..62ef5ca 100644
--- a/thinkpad_ec.c
+++ b/thinkpad_ec.c
@@ -90,8 +90,10 @@ static u64 prefetch_jiffies;                      /* time of prefetch, or: */
 /* Locking: */
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,37)
 static DECLARE_MUTEX(thinkpad_ec_mutex);
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(6,4,0)
 static DEFINE_SEMAPHORE(thinkpad_ec_mutex);
+#else
+static DEFINE_SEMAPHORE(thinkpad_ec_mutex, 1);
 #endif
 
 /* Kludge in case the ACPI DSDT reserves the ports we need. */
diff --git a/tp_smapi.c b/tp_smapi.c
index 6346287..9feebab 100644
--- a/tp_smapi.c
+++ b/tp_smapi.c
@@ -111,8 +111,10 @@ static unsigned short smapi_port;  /* APM control port, normally 0xB2 */
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,37)
 static DECLARE_MUTEX(smapi_mutex);
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(6,4,0)
 static DEFINE_SEMAPHORE(smapi_mutex);
+#else
+static DEFINE_SEMAPHORE(smapi_mutex, 1);
 #endif
 
 /**
